cmd/kube-proxy/app/server_others.go

```diff
@@ -145,6 +152,40 @@ func newProxyServer(
 		return nil, fmt.Errorf("cannot determine detect-local-mode: %v", err)
 	}
 
+	var nodeInfo *v1.Node
+	if detectLocalMode == proxyconfigapi.LocalModeNodeCIDR {
+		klog.Infof("Watching for node %s, awaiting podCIDR allocation", hostname)
+
+		// since allocators can assign the podCIDR after the node registers, we do a watch here to wait
+		// for podCIDR to be assigned, instead of assuming that the Get() on startup will have it.
+		w, err := client.CoreV1().Nodes().Watch(context.TODO(), metav1.ListOptions{FieldSelector: fmt.Sprintf("metadata.name=%s", hostname)})
+		if err != nil {
+			klog.V(0).Infof("Error setting watch on node: %v", err)
+			return nil, fmt.Errorf("unable to set watch on node to configure detect-local-mode %v", proxyconfigapi.LocalModeNodeCIDR)
+		}
+		timeout := time.NewTimer(timeoutForNodePodCIDR)
+	WaitForPodCIDR:
+		for {
+			select {
+			case event := <-w.ResultChan():
```

Comment:
this doesn't handle a closed channel correctly and will hotloop on it

Reply:
Good catch. Will handle that.

Reply:
with the new API, this is not needed as it's handled by the wrapper call.